packages/components/MenuButton/src/MenuButton.tsx

```diff
@@ -82,9 +81,23 @@ export const MenuButton = compose<MenuButtonType>({
         <Slots.button />
         {context.showContextualMenu && (
           <Slots.contextualMenu>
-            {menuItems.map((menuItem) => (
-              <ContextualMenuItem key={menuItem.itemKey} {...menuItem} />
-            ))}
+            {menuItems.map((menuItem) => {
+              const { hasSubmenu, submenu, showSubmenu, componentRef, submenuItems, ...items } = menuItem;
+              return hasSubmenu ? (
+                <Slots.contextualMenuItems>
+                  <SubmenuItem componentRef={componentRef} {...items} />
+                  {showSubmenu && (
+                    <Submenu target={componentRef} {...submenu}>
```

Comment:
This seems like consumers would be limited to just one level of submenu. Is that by design?

Reply:
Or maybe I'm just misreading and nesting submenus is possible

Reply:
In this implementation nested submenues are not supported. But we can discuss, if we need to implement it. I remember from ContextualMenu documentation, that it's better not to use nested Submenues. Here's quote from the docs: "Avoid submenus of submenus as they can be difficult to invoke or remember". @lenahong, what do you think? Should we add opportunity to create nested submenues?

Reply:
If it's by design that's fine. It's probably not a common scenario, so it might not be an issue until a client asks about it.

Reply:
I agree